### PR TITLE
Adding a test for the ExpressionBasedGraphOutputNode graph.

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/test/scala/wdltool/graph/WomDotGraphTest.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        val standardResult = standardiseResult(result)
+        val standardExpectation = standardiseResult(testCase.dotExpectation)
+
+        standardResult should be(standardExpectation)
+      }
+    }
+  }
+
+  def standardiseResult(dot: String) = {
```
Comment (kshakir) [inline src/test/scala/wdltool/graph/WomDotGraphTest.scala]: We don't speak English here. 'murica! 🇺🇸🎆⚾️

**Review by kshakir — APPROVED** [review]

File: src/main/scala/wdltool/graph/WomGraph.scala [hunk 2]
```diff
@@ -96,3 +71,37 @@ object WomGraphPrint {
...
 }
+
+object WomGraph {
+
+  def fromFiles(mainFile: String, auxFiles: Seq[String]): ErrorOr[WomGraph] = {
+    val womExecutable = if (mainFile.toLowerCase().endsWith("wdl")) womExecutableFromWdl(mainFile) else womExecutableFromCwl(mainFile)
+    womExecutable.graph map { graph => new WomGraph(womExecutable.entryPoint.name, graph) }
+  }
+
+  private def readFile(filePath: String): String = Files.readAllLines(Paths.get(filePath)).asScala.mkString(System.lineSeparator())
```
Comment (mcovarr) [inline src/main/scala/wdltool/graph/WomGraph.scala:84]: isn't there already a stdlib function to do this?

File: src/test/scala/wdltool/graph/WomDotGraphTest.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        standardResult should be(standardExpectation)
+      }
+    }
+  }
+  
+  def standardizifyResult(dot: String) = {
+    val regex = "(NODE|PORT)-?[0-9]*".r
+    val matches = regex.findAllIn(dot).toList.distinct.map(_.toString)
+    val currentMatchId = new AtomicInteger(0)
```
Comment (mcovarr) [inline src/test/scala/wdltool/graph/WomDotGraphTest.scala:29]: why must this be atomic?

File: src/test/scala/wdltool/graph/ExpressionBasedGraphOutputNodeSpec.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      |  "PORT5" [shape=oval label="j"];
+      |  "NODE14" [shape=plaintext label="Int x_out"]
+      |}
+      |
+      |}
+      |""".stripMargin
+
+  override val cases = List(WomDotGraphTestCase("ExpressionBasedGraphOutputNodes", expressionOutputGraph, expressionOutputDot))
+
+  tests()
```
Comment (mcovarr) [inline src/test/scala/wdltool/graph/ExpressionBasedGraphOutputNodeSpec.scala:80]: kinda awkward to have to call this at the end...

**Review by mcovarr — APPROVED** [review]

File: src/test/scala/wdltool/graph/WomDotGraphTest.scala (hunk 3 above)
Comment (cjllanwarne) [reply src/test/scala/wdltool/graph/WomDotGraphTest.scala:29]: it doesn't need to be, but it has an `getAndIncrement` method that's nicer than a two line get and set on a regular integer

File: src/test/scala/wdltool/graph/ExpressionBasedGraphOutputNodeSpec.scala (hunk 4 above)
Comment (cjllanwarne) [reply src/test/scala/wdltool/graph/ExpressionBasedGraphOutputNodeSpec.scala:80]: yeah, the trouble was if I didn't do this then I was getting `NPE`s for uninitialized values in the extending classes

**Pull request merged**